src/java.xml/share/classes/com/sun/org/apache/xalan/internal/xsltc/trax/StAXStream2SAX.java

```diff
@@ -281,7 +281,9 @@ private void handleCharacters() throws XMLStreamException {
         int textLength = staxStreamReader.getTextLength();
         char[] chars = new char[textLength];
 
-        staxStreamReader.getTextCharacters(0, chars, 0, textLength);
+        if (textLength > 0) {
+            staxStreamReader.getTextCharacters(0, chars, 0, textLength);
+        }
```

Comment:
Hi Joe,
The comment above this piece reads:
```
        // workaround for bugid 5046319 - switch over to commented section
        // below when it is fixed.
```
And the bug 5046319 (closed issue about AIOOBE) is fixed. Should the commented section be considered?

Reply:
Hi Naoto,

Yes, I did look through that code and bug 5046319. Unfortunately, it happened when JAXP was standalone, the history of the change was lost. 
The variable "buf" seems to be an instance variable that serves to cache the text read. That part had been changed as well. Furthermore, if buf starts with zero length, the call to getTextCharacters would still result in IndexOutOfBoundsException.

Maybe the commented code can be removed, I kept them as is in case they may be useful reference as a history.

Reply:
OK, then I'd suggest removing the above comment and commented section altogether, as it only provides confusion.

Reply:
Yeah, agree. The code is stable as is.